target/riscv/csr.c

```diff
@@ -1347,19 +1343,34 @@ static int read_ccsr(CPURISCVState *env, int csrno, target_ulong *val)
     // The capability cause has moved to xTVAL so we don't report it here.
     RISCVCPU *cpu = env_archcpu(env);
     target_ulong ccsr = 0;
-    ccsr = set_field(ccsr, CCSR_ENABLE, cpu->cfg.ext_cheri);
-    ccsr = set_field(ccsr, CCSR_DIRTY, 1); /* Always report dirty */
+    ccsr = set_field(ccsr, XCCSR_ENABLE, cpu->cfg.ext_cheri);
+    ccsr = set_field(ccsr, XCCSR_DIRTY, 1); /* Always report dirty */
+
+    /* Expose state bits (GCLG) only above U mode */
+    if (env->priv != PRV_U)
+        ccsr |= env->mccsr;
```

Comment:
This doesn't make sense. This would expose the bits in uccsr if you read it from S-mode or M-mode. Also, currently Toooba only has then in sccsr; should they be mirrored in mccsr (like this does) or only in sccsr? I feel Toooba's approach makes more sense? Especially once you start to consider a hccsr for H, then it looks more like satp+hgatp.